### PR TITLE
Cleanup: remove old fields from models

### PR DESCRIPTION
**Pull request by humitos**
I understand that we can remove them all because we needed just to support
`null=` values while doing the deploy.

### CODE REVIEW

**Comment by stsewd** [comment]
We need a data migration for all those models that have a null value, we could also do it from the admin before deploying this change.

**Comment by agjohnson** [comment]
I think I share the sentiment on the data migration, but didn't follow along closely with #7676 at the time either. It seems we'd still have null values in those fields if we don't do a data migration however.

I'm going to pull this one from our release tomorrow, as the data migration is still a question and tests/checks seem angry.

**Comment by humitos** [comment]
I pushed a new commit with the migrations file that defines created/modified as `timezone.now()`. We could use a fixed datetime instead if we prefer, but we don't know _when_ they were created anyways. Let me know if this makes sense to you.

**Comment by humitos** [comment]
@stsewd @agjohnson what do you think about the migration with now as datetime?

File: readthedocs/projects/migrations/0088_timestamp_migration.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    now = timezone.now()
+    query = Q(created__isnull=True) | Q(modified__isnull=True)
+    Project = apps.get_model('projects', 'Project')
+    Project.objects.filter(query).update(
+        created=now,
+        modified=now,
+    )
+
+    ImportedFile = apps.get_model('projects', 'ImportedFile')
+    ImportedFile.objects.filter(query).update(
```
Comment (stsewd) [inline readthedocs/projects/migrations/0088_timestamp_migration.py:18]: The field for this model is `ignore` (boolean), isn't a date that we need to migrate here, similar for Project, the attribute to migrate is `external_builds_privacy_level` (charfield)

**Comment by humitos** [comment]
I updated this PR and I realized that we will have to deal with historical models as well 😕

**Comment by ericholscher** [comment]
Yea... The historical models aren't providing as much value as I hoped they would. I really wish they had a better diffing UX.

**Pull request closed without merging**